cypress/e2e/functional/teacher_tests/teacher_sort_work_view_spec.js

```diff
@@ -66,8 +66,48 @@ describe('SortWorkView Tests', () => {
     sortWork.getSortWorkItem().should('be.visible'); // Verify the document is closed
   });
 
+  it("should open Sort Work tab and test showing by Problem, Investigation, Unit, All", () => {
+    beforeTest(queryParams1);
+
+    sortWork.getShowForMenu().should("be.visible");
+    sortWork.getShowForProblemOption().should("have.class", "selected"); // "Problem" selected by default
+    sortWork.getShowForInvestigationOption().should("exist");
+    sortWork.getShowForUnitOption().should("exist");
+    sortWork.getShowForAllOption().should("exist");
+
+    cy.get(".section-header-arrow").click({multiple: true}); // Open the sections
+    // For the "Problem" option, documents should be listed using the larger thumbnail view
+    cy.get("[data-test=sort-work-list-items]").should("have.length.greaterThan", 0);
+    cy.get("[data-test=simple-document-item]").should("not.exist");
+    sortWork.getShowForMenu().click();
+    cy.wait(500);
+    sortWork.getShowForInvestigationOption().click();
+    cy.wait(500);
+    // For the "Investigation", "Unit", and "All" options, documents should be listed using the smaller "simple" view
+    cy.get("[data-test=sort-work-list-items]").should("not.exist");
+    cy.get("[data-test=simple-document-item]").should("have.length.greaterThan", 0);
+    sortWork.getShowForMenu().click();
+    cy.wait(500);
+    sortWork.getShowForUnitOption().click();
+    cy.wait(500);
+    cy.get("[data-test=sort-work-list-items]").should("not.exist");
+    cy.get("[data-test=simple-document-item]").should("have.length.greaterThan", 0);
+    sortWork.getShowForMenu().click();
+    cy.wait(500);
+    sortWork.getShowForAllOption().click();
+    cy.wait(500);
+    cy.get("[data-test=sort-work-list-items]").should("not.exist");
+    cy.get("[data-test=simple-document-item]").should("have.length.greaterThan", 0);
+    cy.get("[data-test=simple-document-item]").should("have.attr", "title").and("not.be.empty");
+    cy.get("[data-test=simple-document-item]").first().click();
+    sortWork.getFocusDocument().should("be.visible");
+  });
 
-  it("should open Sort Work tab and test sorting by group", () => {
+  // TODO: Reinstate the tests below. They will fail until all metadata documents are updated with
+  // the new fields. Currently, we've only updated a small set of metadata documents in some of the
```

Comment:
Just to clarify, the firebase QA data gets cleared each time. I don't think the firestore data gets cleared at the same time. Because of this each time the test runs it will be creating new documents in firebase. So it isn't possible to pre-generate the metadata documents for the qa data. So we'll need to get the work done which updates the metadata documents in real time, before we can re-enable these tests.

Reply:
Thanks. I didn't realize that. I also underestimated the number of affected tests. The latest commit updates that comment and disables a few more tests.